### PR TITLE
Make it build with MySQL Community Edition 8.0.11

### LINKED ISSUE #6: Does not build with MySQL Community 8.0.11
MySQL Community Edition doesn't have my_bool (since 8.0.1 , I think) and the struct tags are changed too.

I tried to create a PR #4 but the build fails in Travis.
What version of mysql client library is there?

### PR DESCRIPTION
**Pull request by Bu11etmagnet**
Try to fix #6

### CODE REVIEW

File: include/mariadb++/bind.hpp [hunk 1]
```diff
@@ -14,6 +14,10 @@
 #include <mariadb++/types.hpp>
 #include <mariadb++/data.hpp>
 
+#if MYSQL_COMMUNITY_8 > 80000
+typedef bool my_bool;
```
Comment (roussosalex) [inline include/mariadb++/bind.hpp]: Could you move this to types.hpp?

Comment (roussosalex) [inline include/mariadb++/bind.hpp]: Could you move this to types.hpp?

File: include/mariadb++/connection.hpp [hunk 2]
```diff
@@ -18,8 +18,10 @@
 #include <mariadb++/save_point.hpp>
 #include <mariadb++/result_set.hpp>
 
+#if MYSQL_COMMUNITY_8 < 80000
 struct st_mysql;
 typedef struct st_mysql MYSQL;
+#endif
```
Comment (roussosalex) [inline include/mariadb++/connection.hpp]: Is it possible to replace the entire forward declaration by including `mysql/mysql.h`?

File: include/mariadb++/types.hpp [hunk 3]
```diff
@@ -77,4 +78,17 @@ namespace mariadb {
     typedef std::shared_ptr<std::istream> stream_ref;
 }
 
+#ifdef MARIADB_BASE_VERSION
```
Comment (roussosalex) [inline include/mariadb++/types.hpp]: Can you shorten the preprocessor ifs to a single line of `#if defined(MARIADB_BASE_VERSION) && MYSQL_VERSION_ID > foo`

File: include/mariadb++/connection.hpp (hunk 2 above)
Comment (Bu11etmagnet) [reply include/mariadb++/connection.hpp]: Including `mysql/mysql_time.h` should be enough

**Pull request merged**